script/server

```diff
@@ -61,6 +61,22 @@ then
   fatal "config/diaspora.yml is missing! Copy over config/diaspora.yml.example to config/diaspora.yml and edit it properly!"
 fi
 
+# Check if detached head state
+git status | grep 'Changes not staged for commit' &> /dev/null
+if [ $? ]
+then
+  warning "Your changes are not staged for commit. Kindly stash or commit them"
+fi
```

Comment:
You can't detect detached head status by seeing if there are unstaged changes. Detached head status happens when you do for example `git checkout v0.3.0.0`, that is checkout anything other than a branch.

Also the main point was to ensure creating the tarball below doesn't fail in that state.